vortex-btrblocks/src/float.rs

```diff
@@ -0,0 +1,426 @@
+mod dictionary;
+mod stats;
+
+use vortex_alp::{alp_encode, ALPArray, RDEncoder};
+use vortex_array::array::{ConstantArray, PrimitiveArray};
+use vortex_array::variants::PrimitiveArrayTrait;
+use vortex_array::{Array, IntoArray, IntoArrayVariant};
+use vortex_dict::DictArray;
+use vortex_dtype::PType;
+use vortex_error::{vortex_panic, VortexExpect, VortexResult};
+use vortex_runend::compress::runend_encode;
+use vortex_runend::RunEndArray;
+
+use self::stats::FloatStats;
+use crate::float::dictionary::dictionary_encode;
+use crate::integer::{IntCompressor, IntegerStats};
+use crate::{
+    estimate_compression_ratio_with_sampling, integer, Compressor, CompressorStats,
+    GenerateStatsOptions, Scheme,
+};
+
+/// Threshold for the average run length in an array before we consider run-end encoding.
+const RUN_END_THRESHOLD: u32 = 3;
+
+pub trait FloatScheme: Scheme<StatsType = FloatStats, CodeType = FloatCode> {}
+
+impl<T> FloatScheme for T where T: Scheme<StatsType = FloatStats, CodeType = FloatCode> {}
+
+pub struct FloatCompressor;
+
+impl Compressor for FloatCompressor {
+    type ArrayType = PrimitiveArray;
+    type SchemeType = dyn FloatScheme;
+    type StatsType = FloatStats;
+
+    fn schemes() -> &'static [&'static Self::SchemeType] {
+        &[
+            &UncompressedScheme,
```

Comment:
Are you planning Sparse/Frequency as a follow up?

Reply:
I guess the most likely place for it to pop up is for when you have a column of high-precision doubles that uses 0.0 or NaN as an N/A value